src/opencl_xsha512_fmt_plug.c

```diff
@@ -369,7 +369,7 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 		0, NULL, multi_profilingEvent[1]), "Set ND range");
 
 	/// Reset key to unchanged and hashes uncopy to host
-	xsha512_key_changed = 0;
+	new_keys = 0;
     hash_copy_back = 0;
```

Comment:
Perhaps you want to move the `new_keys = 0;` line like you did in `opencl_rawsha512_fmt_plug.c`. And there's the confusing comment about "uncopy" here, too. Please keep these in sync.

Reply:
Fixed now